apps/sms/views/conversation/test/unit/conversation_test.js

```diff
@@ -2598,9 +2599,19 @@ suite('conversation.js >', function() {
     setup(function(done) {
       ConversationView.initializeRendering();
       setActiveThread(8);
-      container =
-        ConversationView.getMessageContainer(testMessages[0].timestamp, false);
+
+      container = ConversationView.getMessageContainer(
+        testMessages[0].timestamp, false
+      );
+
+      this.sinon.stub(MessageManager, 'getMessage').returns(Promise.resolve());
+
+      this.sinon.stub(Threads, 'unregisterMessage', (messageId) => {
+        ConversationView.activeThread.messages.delete(messageId);
+      });
```

Comment:
Well, it's needed since we have ```activeThread``` with non-empty ```messages``` list and once _all_ messages are deleted we have the following ```if``` case:
```
 if (!this.activeThread.messages.size) {
   return this.backOrClose();
 }
```

But now I think it will be better to just clean ```ConversationView.activeThread.messages``` in ```all messages are deleted``` test suite and verify that ```unregisterMessage``` has been called as you suggest.